src/tests/ftest/control/dmg_nvme_scan_test.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/python
+'''
+  (C) Copyright 2018-2019 Intel Corporation.
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+     http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+kworker/16:1
+  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+'''
+from __future__ import print_function
+
+import os
+import time
+
+from dmg_utils import DmgCommand
+from server_utils import ServerCommand
+from apricot import TestWithoutServers
+from avocado.utils import process
+
+class DmgNvmeScanTest(TestWithoutServers):
+    """ Simple test to verify the scan function of the dmg tool.
+    :avocado: recursive
+    """
+    def __init__(self, *args, **kwargs):
+        super(DmgNvmeScanTest, self).__init__(*args, **kwargs)
+
+    @staticmethod
+    def clean_up(self):
+        """ Setup/cleanup for the daos_server to run properly."""
+
+        umount_daos = "umount /mnt/daos; rm -rf /mnt/daos"
+        rm_sockets = "rm -rf /tmp/daos_sockets/"
+        rm_logs = "rm -rf /tmp/*.log"
+
+        try:
+            process.run(
+                umount_daos, verbose=True, ignore_status=True, sudo=True)
+        except Exception:
+            raise process.CmdError(umount_daos)
+
+        try:
+            process.run(
+                rm_sockets, verbose=True, ignore_status=True, sudo=True)
+        except Exception:
+            raise process.CmdError(rm_sockets)
+
+        try:
+            process.run(
+                rm_logs, verbose=True, ignore_status=True, sudo=True)
+        except Exception:
+            raise process.CmdError(rm_logs)
+
+    def test_dmg_nvme_scan_basic(self):
+        """ Test basic dmg functionality to scan nvme the storage on system.
+        :avocado: tags=all,hw,dmg,control
+        """
+        self.clean_up()
```

Comment:
(pylint-no-value-for-parameter) No value for argument 'self' in staticmethod call